### ISSUE TITLE
Printing to a printer connected via USB, fails

### ISSUE THREAD

**Issue by dhfreinkel** [AdoptOpenJDK/openjdk-build#1197]
I am using OpenJ9 with IBM's ACS application.
I am using TN5250 sessions and when I print the screen it fails, basically the application hands off the print to Java and Java will not allow the print to happen via a USB attached printer.

**Comment by karianna** [comment]
@dhfreinkel Can you give us the O/S and version of AdoptOpenJDK you're using?

**Comment by dhfreinkel** [comment]
I am using Windows 10.

JDK-11.0.4+11

 

 

 

Darryl Freinkel

Tel: 770.321.8562 Ext 111 | Cell: 678.355.8562

**Comment by keithc-ca** [comment]
@dhfreinkel I'm not familiar with ACS or TN5250: can you help us reproduce this problem with fewer dependencies? Alternatively, some more details on what you mean by 'allow the print to happen' would be helpful. Does some dialog or error message appear, or does some exception occur that your could share here (with a stack trace)?

**Comment by dhfreinkel** [comment]
Apologies, I should have added more detail.

 

ACS and TN5250 are IBM products that are written in Java and provide many functions to provide access to the IBMi system example TN5250 is a Telnet application.

 

I had switched from Oracle’s Java to OpenJ9 last week. Everything worked great until I tried to do screen prints. The printer is attached as a USB printer. Last week I was on site printing using a network printer and all worked normally.  I work alternatively on site and from home. When at home, I am restricted to the corporate network and can only connect the printer using a USB connection.

 

I worked with the IBM support folk and we could not see the screen prints being sent to the windows control panel printer screen. I was getting the dialog box confirming the printer, layout (portrait/Landscape, size, etc.). Clicking okay is when nothing happened.

 

In the past hour, I tried to downgrade my print driver to a PCL5 driver but Win10 would not permit the older version to be installed. I had deleted the driver on my machine so I had to reinstall the print driver for the HP 2025 printer from HP’s site. After that, I could print the screen prints.

 

At this point I have no idea if it’s a Java issue, IBM issue or a Windows issue. IBM support confirmed they have had a few similar complaints.

 

I am testing your OpenJ9 software for our company which we intend to use in place of Oracle’s Java.

 

I hope this will not recur and will put OpenJ9 on a few other PC’s and test it out.

 

Thank you.  

 

Darryl Freinkel

Tel: 770.321.8562 Ext 111 | Cell: 678.355.8562
